examples/pokemon-service-client-usage/examples/mock-request.rs

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/// This example demonstrates how to create a `smithy-rs` Client using a connector
```

Comment:
Should we just show how to use the existing `capture_request()` method that accomplishes this, or is this illustrating how to implement the `HttpClient`/`HttpConnector` traits?

Reply:
TIL about `capture_request`. Would be nice to have a separate example for it.

Reply:
Oh I didn't know about `capture_request`. I have changed the example now to use `capture_request` instead. Thanks.